src/ADS124S08.cpp

```diff
@@ -64,49 +68,156 @@ void ADS124S08::releaseChipSelect(void){
+	registers[GPIODAT_ADDR_MASK] = 0x00;
+	registers[GPIOCON_ADDR_MASK] = 0x00;
 	fStart = false;
 	releaseChipSelect();
 	deassertStart();
-}
-
-void ADS124S08::begin()
-{
-	SPI.begin( CS_PIN );
-	SPI.setBitOrder(MSBFIRST);
-	SPI.setDataMode( SPI_MODE1 );
-	SPI.setClockSpeed( 1000000 );
-	#if defined (SPI_HAS_TRANSACTION)
-		mySPISettings = SPISettings(1000000, MSBFIRST, SPI_MODE1);
-	#endif
+}
+
+void ADS124S08::SPI_Init(uint32_t SPI_speed, uint8_t chipSelect_Pin)
+{
+	SPI.setMISO(PB14);
+	SPI.setMOSI(PB15);
+	SPI.setSCLK(PB13);
+	SPI.begin();													   //Initiate SPI port
+	SPI.beginTransaction(SPISettings(SPI_speed, MSBFIRST, SPI_MODE0)); //Setup SPI parameters
+	pinMode(chipSelect_Pin, OUTPUT);								   //Set Chip select pin as output
+	digitalWrite(chipSelect_Pin, HIGH);								   //Set Chip select pin high
+	_chipSelect_Pin = chipSelect_Pin;
+}
+
+/* 
+Description: Writes 16bit data to a 16 bit register. 
+Input: Register address, data
+Output:-
+*/
+void ADS124S08::SPI_Write_16(uint16_t Address, uint16_t Data)
+{
+	uint16_t temp_address;
+
+	digitalWrite(_chipSelect_Pin, LOW);
+	temp_address = ((Address << 4) & 0xFFF0); //shift address  to align with cmd packet
+	SPI.transfer16(temp_address);
+	SPI.transfer16(Data);
+
+	digitalWrite(_chipSelect_Pin, HIGH);
+}
+/* 
+Description: Writes 32bit data to a 32 bit register. 
+Input: Register address, data
+Output:-
+*/
+void ADS124S08::SPI_Write_32(uint16_t Address, uint32_t Data)
+{
+	uint16_t temp_address;
+	uint16_t temp_highpacket;
+	uint16_t temp_lowpacket;
+
+	temp_highpacket = (Data & 0xFFFF0000) >> 16;
+	temp_lowpacket = (Data & 0x0000FFFF);
+
+	digitalWrite(_chipSelect_Pin, LOW);
+
+	temp_address = ((Address << 4) & 0xFFF0); //shift address  to align with cmd packet
+	SPI.transfer16(temp_address);
+	SPI.transfer16(temp_highpacket);
+	SPI.transfer16(temp_lowpacket);
+
+	digitalWrite(_chipSelect_Pin, HIGH);
+}
+
+/* 
+Description: Reads 16bit data from register. 
+Input: Register address
+Output: 16 bit data
+*/
+uint16_t ADS124S08::SPI_Read_16(uint16_t Address)
+{
+	uint16_t temp_address;
+	uint16_t returnData;
+
+	digitalWrite(_chipSelect_Pin, LOW);
+
+	temp_address = (((Address << 4) & 0xFFF0) + 8);
+	SPI.transfer16(temp_address);
+	returnData = SPI.transfer16(0);
+
+	digitalWrite(_chipSelect_Pin, HIGH);
+	return returnData;
+}
+
+/* 
+Description: Reads 32bit data from register. 
+Input: Register address
+Output: 32 bit data
+*/
+uint32_t ADS124S08::SPI_Read_32(uint16_t Address)
+{
+	uint16_t temp_address;
+	uint16_t temp_highpacket;
+	uint16_t temp_lowpacket;
+	uint32_t returnData;
+
+	digitalWrite(_chipSelect_Pin, LOW);
+
+	temp_address = (((Address << 4) & 0xFFF0) + 8);
+	SPI.transfer16(temp_address);
+	temp_highpacket = SPI.transfer16(0);
+	temp_lowpacket = SPI.transfer16(0);
+
+	digitalWrite(_chipSelect_Pin, HIGH);
+
+	returnData = temp_highpacket << 16;
+	returnData = returnData + temp_lowpacket;
+
+	return returnData;
+}
+
+void ADS124S08::begin()
+{
+	SPI.setMISO(PB14);

```

Comment:
As mentioned above, PB13-15 only applies to this specific micro. A more flexible approach would be to add code to over overloading of the ::begin method that allows passing these hardware specific pins.